plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -181,6 +182,12 @@ public ConnectorPageSource createPageSource(
         if (filteredSplitPredicate.isNone()) {
             return new EmptyPageSource();
         }
+        Map<DeltaLakeColumnHandle, Domain> partitionColumnDomains = filteredSplitPredicate.getDomains().orElseThrow().entrySet().stream()
```

Comment:
Usually we do partition pruning first. I would move this above line 174 and use `dynamicFilter.getCurrentPredicate().transformKeys(DeltaLakeColumnHandle.class::cast)` instead of `filteredSplitPredicate`.

Reply:
@raunaqmorarka I think this is consistent with Iceberg code node